packages/services/test/integration/notification-scanners/SubscriptionReminder.js

```diff
@@ -0,0 +1,145 @@
+const MINUTES = 60 * 1000
+const HOURS = 60 * MINUTES
+const DAYS = 24 * HOURS
+
+
+describe('SubscriptionReminder notifications', () => {
+
+  after(async () => {
+    await userDbCleanup(TEST_ADDR, TEST_EMAIL)
+    await userNotificationTypeDbCleanup(notificationTypeModel)
+  })
+
+  let logger = {}
+  beforeEach(async () => {
+    logger = {
+      success: sinon.fake(),
+      warn: sinon.fake(),
+    }
+  })
+
+  it('should not create a notification for a user with verified address (opted out)', async () => {
+    await userDbCleanup(TEST_ADDR, TEST_EMAIL)
+    await userNotificationTypeDbCleanup(notificationTypeModel)
+    await User.query().insertGraph({
+      address: TEST_ADDR,
+      addressVerified: true,
+      emailVerified: false,
+      email: {
+        email: TEST_EMAIL
+      }
+    })
+    await tryRunScanner(logger, notificationTypeModel)
+    const type = await userNotificationTypeByModel(notificationTypeModel)
+    expect(type.notifications.length).to.equal(0)
+    expect(logger.success).to.have.callCount(1)
+  })
+
+  it('should not create a notification for a user with a recent verification token', async () => {
+    await userDbCleanup(TEST_ADDR, TEST_EMAIL)
+    await userNotificationTypeDbCleanup(notificationTypeModel)
+    await User.query().insertGraph({
+      address: TEST_ADDR,
+      addressVerified: true,
+      emailVerified: false,
+      email: {
+        email: TEST_EMAIL
+      },
+      emailVerificationToken: {
+        token: 'test',
+        email: TEST_EMAIL,
+        expiresAt: new Date()
+      }
+    })
+    await tryRunScanner(logger, notificationTypeModel)
+    const type = await userNotificationTypeByModel(notificationTypeModel)
+    expect(type.notifications.length).to.equal(0)
+    expect(logger.success).to.have.callCount(1)
+  })
+
+  it('should create a notification for a user with a day old verification token', async () => {
+    await userDbCleanup(TEST_ADDR, TEST_EMAIL)
+    await userNotificationTypeDbCleanup(notificationTypeModel)
+    const user = await User.query().insertGraph({
+      address: TEST_ADDR,
+      addressVerified: true,
+      emailVerified: false,
+      email: {
+        email: TEST_EMAIL
+      },
+      emailVerificationToken: {
+        token: 'test',
+        email: TEST_EMAIL,
+        expiresAt: new Date(Date.now()-DAYS)
+      }
+    })
+    await tryRunScanner(logger, notificationTypeModel)
+    const type = await userNotificationTypeByModel(notificationTypeModel)
+    expect(type.notifications.length).to.equal(1)
+    expect(type.notifications[0].details).to.deep.equal({
+      emailTemplateModel: {
+        emailPreferencesUrl: CLIENT_URL
+      },
+      token: user.emailVerificationToken.id
+    })
+    expect(logger.success).to.have.callCount(1)
+  })
+
+  it('should not run scan again before scan period', async () => {
+    await tryRunScanner(logger, notificationTypeModel)
+    expect(logger.success).to.have.callCount(0)
```

Comment:
This test, and some others relying on the logger being called, are a bit brittle, but not sure if there's something better we can do with a spy instead.

I mention this one especially, because it would pass even if `tryRunScanner()` was completely broken and did nothing.

Reply:
I couldn't think of better spy function.
For that specific test, `tryRunScanner` is supposed to do nothing and return. If there was some bad error then we would see an exception and the whole test would fail